src/Compilers/Core/Portable/Collections/SmallDictionary.cs

```diff
@@ -276,7 +276,9 @@ private void Insert(int hashCode, K key, V value, bool add)
                 {
                     if (currentNode.Left == null)
                     {
-                        currentNode.Left = currentNode = new AvlNode(hashCode, key, value);
+                        var previousNode = currentNode;
```

Comment:
This fixes violation of a newly added rule (CA2246) for improved clarity and readability: `Assigning to a symbol and its member (field/property) in the same statement is not recommended. It is not clear if the member access was intended to use symbol's old value prior to the assignment or new value from the assignment in this statement. For clarity, consider splitting the assignments into separate statements.`

We have few options here:
1. Keep the current code change and keep this rule enabled.
2. Revert the code changes in this file and disable the rule for the repo.
3. Apply inline suppressions here if we feel the rule is useful as such but the prior code style for this case is more desirable.

Reply:
I think this is a good change. But someone from @dotnet/roslyn-compiler needs to sign off.

Reply:
This is goodness.

Reply:
This is goodness. Case in point I was halfway through rejecting the PR because I thought you had re-ordered the effects incorrectly. Had to remind myself of the subtle order in how `currentNode` is evaluated before I could convince myself this was correct. New code is much easier to grok.

Reply:
I agree. Thanks to @maxkoshevoi for contributing the new rule.

Reply:
Glad I could help.